src/main/java/ch/njol/skript/expressions/ExprBarterDrops.java

```diff
@@ -0,0 +1,152 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.PiglinBarterEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+@Name("Barter Drops")
+@Description("The items dropped by the piglin in a piglin bartering event.")
+@Examples({
+	"on piglin barter:",
+		"\tif the bartering drops contain a jack-o-lantern:",
+			"\t\tremove jack-o-lantern from bartering output",
+			"\t\tbroadcast \"it's not halloween yet!\""
+})
+@Since("INSERT VERSION")
+public class ExprBarterDrops extends SimpleExpression<ItemType> {
+	
+	static {
+		Skript.registerExpression(ExprBarterDrops.class, ItemType.class,
+				ExpressionType.SIMPLE, "[the] [piglin] barter[ing] drops");
+	}
+
+	private Kleenean delay;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern,
+						Kleenean isDelayed, ParseResult result) {
+		if (!getParser().isCurrentEvent(PiglinBarterEvent.class)) {
+			Skript.error("The expression 'barter drops' can only be used in the piglin bartering event");
+			return false;
+		}
+
+		delay = isDelayed;
+
+		return true;
+	}
+	
+	@Override
+	@Nullable
+	protected ItemType[] get(Event event) {
+		if (!(event instanceof PiglinBarterEvent))
+			return null;
+
+		return ((PiglinBarterEvent) event).getOutcome()
+			.stream()
+			.map(ItemType::new)
+			.toArray(ItemType[]::new);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (delay != Kleenean.FALSE) {
+			Skript.error("Can't change the piglin bartering drops after the event has already passed");
```

Comment:
```suggestion
			Skript.error("Cannot modify piglin bartering drops; the event has already occurred.");
```
personally didnt like the previous error message, this looks nicer.

Reply:
The previous error message is in line with existing messages for similar errors.